docs/guides/vault/restrict-addresses-to-vaults.md

```diff
@@ -15,61 +18,363 @@ This way, when **granting a third party permissions** to execute through your pr
+import { ethers } from 'ethers';
+
+// constants
+const provider = new ethers.JsonRpcProvider('https://rpc.l16.lukso.network');
+const universalProfileAddress = '0x..'; // address of the UP
+const vaultAddress = '0x..'; // address of the Vault
 const thirdPartyAddress = '0x..'; // address of the third party you want to restrict
 
+// setup your EOA
+const privateKey = '0x...'; // your EOA private key (controller address)
+const myEOA = new ethers.Wallet(privateKey).connect(provider);
+```
+
+  </TabItem>
+
+</Tabs>
+
+## Step 2 - Create contract instances
+
+At this point we will create instances for the following contracts:
+
+- [**Universal Profile**](../../standards/universal-profile/lsp0-erc725account.md)
+- [**Key Manager**](../../standards/universal-profile/lsp6-key-manager.md)
+
+<Tabs>
+  
+  <TabItem value="web3js" label="web3.js">
+
+```typescript title="Universal Profile & Key Manager contract instances"
 // create an instance of the UP
-const myUP = new web3.eth.Contract(
+const universalProfile = new web3.eth.Contract(
   UniversalProfile.abi,
-  myUniversalProfileAddress,
+  universalProfileAddress,
 );
 
-const allowedAddressesDataKey = // constructing the data key of allowed addresses
-  ERC725YDataKeys.LSP6['AddressPermissions:AllowedAddresses'] +
+// getting the Key Manager address from UP
+const keyManagerAddress = await universalProfile.methods.owner().call();
+// create an instance of the KeyManager
+const keyManager = new web3.eth.Contract(LSP6KeyManager.abi, keyManagerAddress);
+```
+
+  </TabItem>
+
+  <TabItem value="ethersjs" label="ethers.js">
+
+```typescript title="Universal Profile & Key Manager contract instances"
+// create an instance of the UP
+const universalProfile = new ethers.Contract(
+  universalProfileAddress,
+  UniversalProfile.abi,
+);
+
+// getting the Key Manager address from UP
+const keyManagerAddress = await universalProfile.owner();
+// create an instance of the KeyManager
+const keyManager = new ethers.Contract(keyManagerAddress, LSP6KeyManager.abi);
+```
+
+  </TabItem>
+
+</Tabs>
+
+## Step 3 - Encode the calldata for encoding [`AllowedCalls`](../../standards/universal-profile/lsp6-key-manager.md#allowed-calls)
+
+Now we need to encode the **Allowed Calls** that we want for the _Third Party address_. After we do that, we will encode a calldata that will update the _Allowed Calls data key_ with the encoded **Allowed Calls**
+
+<Tabs>
+  
+  <TabItem value="web3js" label="web3.js">
+
+```typescript title="Calldata that will update the Allowed Calls of a Controller address"
+const allowedCallsDataKey = // constructing the data key of allowed addresses
+  ERC725YDataKeys.LSP6['AddressPermissions:AllowedCalls'] +
   thirdPartyAddress.substring(2); // of the 3rd party
 
-// the data value holding the addresses that the 3rd party is allowed to interact with
-const arrayOfAddresses = web3.eth.abi.encodeParameter('address[]', [
-  myVaultAddress,
+const allowedCallsSchema = {
```

Comment:
could this be imported from erc725.js instead?

Reply:
Hmm good question. I am actually don't know if you can and how. If you have something in mind, please send me :simle: